using_restricted_array.rb

```diff
@@ -7,41 +7,236 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  # if array is empty done
+  # set i to 0
+  # while i is less than array length
+  # - increment i by 1
+  # outside loop, return i
+  i = 0
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  # we know the length of the array
+  # it contains integers
+  # want: print each integer value in array
+
+  # set variable to array's length
+  # loop array by array's length times
+  # - print value at each index
+
+
+
+  length(array).times do |value|
+    print "#{array[value]}, "
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+=begin
+1. We know length of array
+2. it contains integets
+3. it is unsorted
+
+Want: find a known value in array
+once found return true
+return false if not found.
+
+Algorithm
+1. set i to 0
+2. while i < length
+  - check value at i
+  - if a match return true
+  - increment i by 1
+3. outside loop return false
+=end
+
+  i = 0
+
+  while i < length
+    if array[i] == value_to_find
+
+      return true
+    end
+    i += 1
+    return false
+  end
+
+
+  # Another Algorithm
+  # 1. interat over the array length times
+  # 2. if value_to_find matchs value at index, return true.
+  # 3. otherwise return false, outside loop
+  #
+  # length.times do |value|
+  #   if value == value_to_find
+  #     return true
+  #   end
+  # end
+  # return false
 end
 
+
 # Finds and returns the largest integer value the array
 # Assumes that the array is not sorted.
 def find_largest(array, length)
-  puts "NOT IMPLEMENTED"
+=begin
+  1. We know length of array
+  2. it contains integets
+  3. it is unsorted
+
+  Want:
+  1. find max value in array
+  2. once found return max
+
+
+  Algorithm
+  1. set i to 0
+  2. set max to value at i
+  3. while i < length
+    - if value at i > max
+      -  replace max with value at i
+    - increment i by 1
+  4.  return max
+=end
+
+  i = 0
+  max = array[0]
+
+  #
+  # while i < length(array)
```

Comment:
All the code seems commented out. I don't think this version is working.

Reply:
The comment code should work though.

Reply:
Using different types of loop doesn't change the algorithm. You can leverage any of the different loop choices available in a language.